src/cli.rs

```diff
@@ -46,14 +46,19 @@ impl Cli {
 
     /// Runs the CLI.
     pub fn run(&self) -> crate::Result {
-        if self.should_edit_file() {
-            self.edit_file()?;
-            return Ok(());
+        // Early return for edit mode
+        if let Some(edit_config) = self.edit_config {
+            return self.edit_file(edit_config);
         }
 
+        self.run_tree()
+    }
+
+    /// Runs the main tree functionality.
+    fn run_tree(&self) -> crate::Result {
         let git = Git::new(&self.path).expect("Should be able to read the git repository");
 
-        // NOTE The Lua state must live as long as the configuration values.
+        // Build Lua state with git if available
```

Comment:
```suggestion
        // NOTE The Lua state must live as long as the configuration values.
```
Why delete this note? It helps explain that `mlua` can act in surprising ways for Rust users. For example, if a `mlua::Function` lives longer than the `mlua::Lua` that created it, calling that function could panic.